cypress/e2e/po/pages/explorer/workloads/workloads.po.ts

```diff
@@ -0,0 +1,182 @@
+import PagePo from '@/cypress/e2e/po/pages/page.po';
+import BaseResourceList from '@/cypress/e2e/po/lists/base-resource-list.po';
+import LabeledInputPo from '@/cypress/e2e/po/components/labeled-input.po';
+import AsyncButtonPo from '@/cypress/e2e/po/components/async-button.po';
+import LabeledSelectPo from '@/cypress/e2e/po/components/labeled-select.po';
+import WorkloadPo from '@/cypress/e2e/po/pages/explorer/workloads.po';
+import PromptRemove from '~/cypress/e2e/po/prompts/promptRemove.po';
+
+export class workloadDetailsPageBasePo extends PagePo {
+  static url: string;
+
+  private static createPath(
+    workloadId: string,
+    clusterId: string,
+    workloadType: string,
+    namespaceId: string,
+    queryParams?: { [key: string]: string }
+  ) {
+    const urlStr = `/c/${ clusterId }/explorer/${ workloadType }/${ namespaceId }/${ workloadId }`;
+
+    if (!queryParams) {
+      return urlStr;
+    }
+
+    const params = new URLSearchParams(queryParams);
+
+    return `${ urlStr }?${ params.toString() }`;
+  }
+
+  static goTo(): Cypress.Chainable<Cypress.AUTWindow> {
+    return super.goTo(this.url);
+  }
+
+  constructor(workloadId: string, clusterId: string, workloadType: string, queryParams?: { [key: string]: string }, namespaceId = 'default') {
+    super(workloadDetailsPageBasePo.createPath(workloadId, clusterId, workloadType, namespaceId, queryParams));
+
+    workloadDetailsPageBasePo.url = workloadDetailsPageBasePo.createPath(
```

Comment:
Is there not a way to pass it as a whole object, so you don't have to define it like 4 times the keys and 2 times the type?
You may create a type out of that as well.